docs/modules/ROOT/pages/getting-started/filters-and-projections.adoc

```diff
@@ -0,0 +1,191 @@
+
+
+== Filtering
+
+When creating a `MATCH` clause, it is possible to add subclauses to it in a similar way to how you added the `RETURN` statement. Change the `Cypher.Match` instance to add a `WHERE` subclause with the method `.where`:
+
+```javascript
+const clause = new Cypher.Match(movieNode)
+    .where(Cypher.eq(movieNode.property("title"), new Cypher.Param("The Matrix")))
+    .return(movieNode);
+```
+
+There is quite a bit to unwrap here. First, you'll notice that the `.where` method takes the result of the function `Cypher.eq`. This function is the **eq**uality (`=`) operator that compares the property `title` and the string `The Matrix`.
+
+`Cypher.eq` takes two arguments to compare. Our first argument will be the property `title`. Because our title is a property of our movieNode variable, the `.property` method of that node will ensure that the property correctly references the movieNode variable.
+
+The second parameter is `"The Matrix"`, but instead of using a plain string, the object `Cypher.Param` class will take care of turning that into a parameter so we can safely put user input here.
+
+
+.Why `Cypher.eq` doesn't need `new`?
+[NOTE]
+====
+Cypher Builder treats Variables (like nodes and relationships) and Clauses as classes, as these are intended to be assigned to a JavaScript variable to be reused, composed and built.
+Cypher Functions and Procedures, however, are intended to be used a single time, as part of a larger query, so Cypher Builder treats these as simple functions, making it closer to how these work in Cypher.
+====
+
+=== Cypher and params
+
+You can run your script again (`node main.js`), and should get the following Cypher:
+
+```cypher
+MATCH (this0:Movie)
+WHERE this0.title = $param0
+RETURN this0
+```
+
+Success! The movie is now filtered by its title to a param. The param, however, is missing from the output. You'll need to tweak the build process at the end of the script to get the parameters as well:
+
+```javascript
+const { cypher, params } = match.build();
+console.log(cypher);
+console.log(params);
+```
+
+Now, executing again, we get all we need to run this query in Neo4j:
+
+```cypher
+MATCH (this0:Movie)
+WHERE this0.title = $param0
+RETURN this0
+```
+
+```javascript
+{ param0: 'The Matrix' }
+```
+
+[TIP]
+====
+Note that `new Param` already takes care of assigning a param name and correctly matching the value to that parameter in the Cypher query. If a `Param` instance is used multiple times within a query, it will always reference the same parameter (`$param0` in this case).
+====
+
+=== `AND` filtering
+
+The query still lacks the second filter, which needs to be concatenated with `AND`. Simply use the method `.and` after the `where` with a new `Cypher.eq` predicate:
+
+```javascript
+    .where(Cypher.eq(movieNode.property("title"), new Cypher.Param("The Matrix")))
+    .and(Cypher.lt(movieNode.property("released"), new Cypher.Param(2000)))
```

Comment:
I believe this could be opinionated, maybe @lidiazuin  knows better here, but I believe that providing these steps at this level of isolation increases the cognitive complexity for the reader, also a more experienced dev who comes from others SQL/Cypher builders will use the getting started as a fast overview of the product without actually have an incremental script file in front. What I'm trying to say is that maybe it's better to not break the method chain in the step, for instance:

```suggestion
const clause = new Cypher.Match(movieNode)
  .where(Cypher.eq(movieNode.property("title"), new Cypher.Param("The Matrix")))
  .and(Cypher.lt(movieNode.property("released"), new Cypher.Param(2000)))
  .return(movieNode);
```

Reply:
Well, the getting started part is mostly for people who are just getting to start using a feature, so it's good to have more explanation and contextualization. I just don't know if that is going too much into detail or not, if we indeed need to break up a little bit to make things more understandable to more inexperienced readers.

Reply:
Here I agree with @lidiazuin
At least my intention here was to bring someone with little knowledge to be able to use the library and it's main features. Other guides like how-tos should definitely be more concise IMO

Reply:
If we stick to the definition of guides and tutorials proposed by Diataxis, we can understand that tutorials are designed for new users who need to learn what's all about and a guide is for a more experienced reader. A comparison would be a tutorial on how to start cooking and a guide is a recipe for chocolate cake 😊

Reply:
Sure, anyway it's already a lot of effort writing these kinds of documentation, so read my comment as a genuine curiosity about the style and not as a request for change!

Reply:
The sentiment behind the comment was that this guide is structured in the form of:
- incremental step
- incremental step
- ...incremental step
- conclusion.

**For a very beginner dev**: The only positive feedback happens at the conclusion step. If the user made a mistake, he needs to reach the conclusion step to recognize it.

**For a very expert user**: He already started with the expectation to find something along `Match(entity).where(predicate)` or `Select(columns).from(entity).build()`, so potentially (**bad** 🙊 ) immediately jump to the conclusion section, with the risk to missing some interesting catch.

But is a comment completely driven by sentiment, so happy if I'm wrong!